src/POData/UriProcessor/UriProcessorNew.php

```diff
@@ -331,6 +330,42 @@ protected function executePost()
                 }
 
                 $payload = $this->getRequest()->getData();
+                if ($payload instanceof ODataURL) {
+                    try {
+                        $this->executeGet();
+                        //dd($this->getRequest()->getSegments()[1]);
+                        $masterModel = $this->getRequest()->getSegments()[0]->getResult();
+                        $masterResourceSet = $this->getRequest()->getSegments()[0]->getTargetResourceSetWrapper();
+                        $masterNavProperty = $this->getRequest()->getLastSegment()->getIdentifier();
+                        $slaveModelUri = new \POData\Common\Url($payload->url);
+                        $host = $this->service->getHost();
+                        $absoluteServiceUri = $host->getAbsoluteServiceUri();
+                        $requestUriSegments = array_slice(
+                        $slaveModelUri->getSegments(),
+                        $absoluteServiceUri->getSegmentCount()
+                    );
+                        $newSegments = \POData\UriProcessor\ResourcePathProcessor\SegmentParser\SegmentParser::parseRequestUriSegments(
+                        $requestUriSegments,
+                        $this->service->getProvidersWrapper(),
+                        true
+                    );
+                        $this->executeGetResource($newSegments[0]);
+                        $slaveModel = $newSegments[0]->getResult();
+                        $slaveResourceSet = $newSegments[0]->getTargetResourceSetWrapper();
+                        $linkAdded = $this->getProviders()->hookSingleModel($masterResourceSet, $masterModel, $slaveResourceSet, $slaveModel, $masterNavProperty);
+                        if ($linkAdded) {
+                            $this->getService()->getHost()->setResponseStatusCode(HttpStatus::CODE_NOCONTENT);
+                        } else {
+                            throw ODataException::createInternalServerError('AdapterInidicatedLinkNotAttached');
+                        }
+                    } catch (\Exception $e) {
+                        dd($e);
```

Comment:
Missed during cleanup?

Reply:
missed cleaning up the try catch in general, the try, catch and dd can all go.